core/src/main/java/org/elasticsearch/index/translog/TranslogWriter.java

```diff
@@ -118,6 +128,18 @@ public static Type fromString(String type) {
         }
     }
 
+    protected final void closeWithTragicEvent(Throwable throwable) throws IOException {
+        try (ReleasableLock lock = writeLock.acquire()) {
+            if (throwable != null) {
```

Comment:
Is the incoming throwable really allowed to be null sometimes?  Seems dangerous because then we could close ourselves here but `.getTragicException()` would return null, as if no tragedy occurred?